lib/middleware.py

```diff
@@ -0,0 +1,31 @@
+from django.conf import settings
+from django.conf.urls.i18n import is_language_prefix_patterns_used
+from django.http import HttpResponseRedirect
+from django.utils import translation
+from django.utils.cache import patch_vary_headers
+from django.utils.deprecation import MiddlewareMixin
+
+
+class LocaleMiddleware(MiddlewareMixin):
+
+    response_redirect_class = HttpResponseRedirect
+
+    def process_request(self, request):
+        urlconf = getattr(request, 'urlconf', settings.ROOT_URLCONF)
+        i18n_patterns_used, prefixed_default_language = is_language_prefix_patterns_used(urlconf)
```

Comment:
I think you can drop the language prefix / path stuff from the process_request too